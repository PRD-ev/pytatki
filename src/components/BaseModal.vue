<template>
  <div>
    <div @click="openModal">
      <slot name="trigger"></slot>
    </div>
    <div v-if="modalOn">
      <div @click="closeModal" class="dark-background"></div>
      <div class="modal">
        <div @click="closeModal" class="close-modal">
          <img src="@/assets/icons/close.svg" alt="close modal">
        </div>
        <slot name="modal-content"></slot>
      </div>
    </div>
  </div>
</template>


<script>
import Vue from 'vue';

export default Vue.extend({
  name: 'BaseModal',
  data() {
    return {
      modalOn: false,
    };
  },
  methods: {
    openModal() {
      this.modalOn = true;
    },
    closeModal() {
      this.modalOn = false;
    },
  },
});
</script>


<style lang="scss" scoped>
.close-modal {
  filter: invert(1);
  position: absolute;
  top: 30px;
  right: 30px;
  cursor: pointer;
}
.dark-background {
  background-color: #000000be;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 5;
  cursor: pointer;
}
.modal {
  z-index: 6;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: var(--white);
  padding: 40px 50px;
  border-radius: 10px;
  max-height: 90vh;
  max-width: 90vw;
  overflow: auto;
}
</style>
