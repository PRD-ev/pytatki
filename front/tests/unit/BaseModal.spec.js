import { shallowMount } from '@vue/test-utils';
import BaseModal from '@/components/BaseModal.vue';

describe('BaseModal.vue', () => {
  it('Opens correctly', () => {
    const wrapper = shallowMount(BaseModal, {
      slots: {
        trigger: '<p class="open"></p>',
        'modal-content': "<p>Hi, I'm the content</p>",
      },
    });
    wrapper.find('.open').trigger('click');
    expect(wrapper.vm.modalOn).toBe(true);
  });

  it('Closes correctly', () => {
    const wrapper = shallowMount(BaseModal, {
      slots: {
        trigger: '<p class="open"></p>',
        'modal-content': "<p>Hi, I'm the content</p>",
      },
    });
    wrapper.setData({ modalOn: true });
    wrapper.find('.close-modal').trigger('click');
    expect(wrapper.vm.modalOn).toBe(false);
  });

  it('Displays content when opened', () => {
    const wrapper = shallowMount(BaseModal, {
      slots: {
        trigger: '<p class="open"></p>',
        'modal-content': '<p class="content"></p>',
      },
    });
    wrapper.setData({ modalOn: true });
    expect(wrapper.find('.content').exists()).toBe(true);
  });

  it('Hides content when closed', () => {
    const wrapper = shallowMount(BaseModal, {
      slots: {
        trigger: '<p class="open"></p>',
        'modal-content': '<p class="content"></p>',
      },
    });
    expect(wrapper.find('.content').exists()).toBe(false);
  });

  it('Matches snapshot', () => {
    const wrapper = shallowMount(BaseModal, {
      slots: {
        trigger: '<p class="open"></p>',
        'modal-content': '<p class="content"></p>',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
