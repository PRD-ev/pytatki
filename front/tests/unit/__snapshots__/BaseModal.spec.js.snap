// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseModal.vue Matches snapshot 1`] = `
<div>
  <div>
    <p class="open"></p>
  </div>
  <!---->
</div>
`;
